errors/errors.go

```diff
@@ -321,3 +334,44 @@ func (e *wrappedError) Format(s fmt.State, verb rune) {
 		writeSimpleFrame(s, stack[0])
 	}
 }
+
+// HasErrorCode checks if this error would return the named error code
+func HasErrorCode(err error, code uint32) bool {
```

Comment:
The general form now is `errors.Is`, correct.
If `HasErrorCode` is only used by Redact, please remove it and upgrade the implementation of Redact

Reply:
@ethanfrey it's not entirely true, because errors.Is will fail for internal errors if it's not the same reference AFAIK. So HasErrorCode is useful still?

Reply:
We can make it package-private though

Reply:
I will make it package-private and merge